pypeit/core/telluric.py

```diff
@@ -189,6 +189,8 @@ def read_telluric_pca(filename, wave_min=None, wave_max=None, pad_frac=0.10):
 
     dwave, dloglam, resln_guess, pix_per_sigma = wvutils.get_sampling(wave_grid)
     tell_pad_pix = int(np.ceil(10.0 * pix_per_sigma))
+    if dloglam == 0.0:
+        msgs.error('The telluric PCA method has zero spacing in log wavelength. This is not supported.')
```

Comment:
Can we move these lines into the `wvutils.get_sampling` function?  The pro is that we don't have to repeat these lines so many time; the cons are that the error message might be less specific and there may be some application of `wvutils.get_sampling` where things can proceed, even if `dloglam==0` (although I don't know what that would be).